detox/src/artifacts/screenshot/SimulatorScreenshotPlugin.js

```diff
@@ -51,19 +53,21 @@ class SimulatorScreenshotPlugin extends ScreenshotArtifactPlugin {
   }
 
   _onInvokeFailure({ params }) {
-    const artifacts = {
-      visibilityFailingScreenshots: params.visibilityFailingScreenshotsURL,
-      visibilityFailingRects: params.visibilityFailingRectsURL,
-    };
+    const visibilityArtifactDirs = [
+      params.visibilityFailingScreenshotsURL,
+      params.visibilityFailingRectsURL
+    ].filter(Boolean);
 
-    for (const [key, artifactPath] of Object.entries(artifacts)) {
-      if (!artifactPath || fs.isDirEmptySync(artifactPath)) {
+    for (const visiblityDir of visibilityArtifactDirs) {
+      if (fs.isDirEmptySync(visiblityDir)) {
         continue;
       }
 
-      this._registerSnapshot(`${key}/`, new FileArtifact({
-        temporaryPath: artifactPath,
-      }));
+      for (const innerPath of fs.globSync(visiblityDir, '*.png')) {
+        this._registerSnapshot(innerPath.replace(/ /g, '_'), new FileArtifact({
```

Comment:
Why to specifically fail-fast on `isDirEmptySync`? What if there are several files but their extension is not `png`?

Since you read the content of the directory twice, in `isDirEmptySync` and in the `globSync` call, I suggest to refactor it so it will read the contents only once.

```js
files = fs.readdirSync(visiblityDir);
files.forEach(file => {
  if (path.extname(file) != ".png") {
    continue;
  }
...
```

Reply:
@asafkorem, the plugin does not really know how to work with non-PNGs – I hope you will communicate to me the upcoming changes if you'll be changing something there on iOS side. So, here I just limit the plugin to what we really have tested, without surprises.

Reply:
I'll follow the suggestion, thanks. See you in the next iteration of this pull request.